4501 remove pagination links logic and references in docs

cadibemma opened the pull request:
## Summary

- Removed 'links' object from paginated endpoints. Removed the existing links construction logic, remove it from tests, and remove it from the docs.

## Related issue(s)
- department-of-veterans-affairs/va-mobile-app#4501


## Testing done

- Removed tests that corresponds to removed code



## What areas of the site does it impact?
- None

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

kpethtel commented on modules/mobile/docs/schemas/Appointments.yml:
```diff
@@ -2,7 +2,6 @@ type: object
 required:
   - data
   - meta
-  - links
```
Seems like we're pretty inconsistent in how we handle this. Sometimes we added links, sometimes not. Sometimes we add meta, sometimes not. Maybe worth a ticket to just double check and clean up this documentation since things are pretty quiet right now.

kpethtel commented on modules/mobile/spec/request/claims_and_appeals_overview_request_spec.rb:
```diff
@@ -79,7 +79,6 @@
             expect(response).to have_http_status(:ok)
             # check a couple entries to make sure the data is correct
             parsed_response_contents = response.parsed_body['data']
-            expect(response.parsed_body.dig('links', 'next')).to be(nil)
```
Good catch. This type of test would've been easy to miss.

kpethtel reviewed (changes requested):
This looks good, but you also need to run generate_docs and add the index.html. 

You can also remove references to `optional_params` and `url` from the pagination helper. They're not being used.

Also, I'd suggest having Narin look at this before we merge just to ensure that he understood what we were talking about and can confirm that this will not cause issues.

cadibemma commented:
> This looks good, but you also need to run generate_docs and add the index.html.
> 
> You can also remove references to `optional_params` and `url` from the pagination helper. They're not being used.
> 
> Also, I'd suggest having Narin look at this before we merge just to ensure that he understood what we were talking about and can confirm that this will not cause issues.

Got it. Updated the index.html and removed references to `optional_params` and `url`

kpethtel reviewed (approved):
Looks good! 👍

kpethtel reviewed (approved):
Looks good!
